__tests__/pet.test.js

```diff
@@ -0,0 +1,149 @@
+const Pet = require('../src/pet');
+
+let pet;
+beforeEach(() => {
+	pet = new Pet('Fido');
+});
+describe('constructor', () => {
+	it('returns an object', () => {
+		expect(new Pet('Fido')).toBeInstanceOf(Object);
+	});
+	it('sets pet name to parameter name', () => {
+		expect(pet.name).toEqual('Fido');
+	});
+	it('sets inital age to 0', () => {
+		expect(pet.age).toEqual(0);
+	});
+	it('sets inital hunger to 0', () => {
+		expect(pet.hunger).toEqual(0);
+	});
+	it('sets initial fitness to 10', () => {
+		expect(pet.fitness).toEqual(10);
+	});
+});
+
+describe('growUp', () => {
+	it('increments the pet age by one', () => {
+		pet.growUp();
+		expect(pet.age).toEqual(1);
+	});
+	it('increases the pet hunger by 5', () => {
+		pet.growUp();
+		expect(pet.hunger).toEqual(5);
+	});
+	it('decreases the pet fitness by 3', () => {
+		pet.growUp();
+		expect(pet.fitness).toEqual(7);
+	});
+	it('throws an error if the pet is not alive', () => {
+		pet.age = 30;
+		expect(() => pet.growUp()).toThrow('Your pet is no longer alive :(');
+	});
+});
+
+describe('walk', () => {
+	it('increases fitness by 4', () => {
+		pet.fitness = 5;
+		pet.walk();
+		expect(pet.fitness).toEqual(9);
+	});
+	it('stops fitness going over 10', () => {
+		pet.fitness = 9;
+		pet.walk();
+		expect(pet.fitness).toEqual(10);
+	});
+	it('throws an error if the pet is not alive', () => {
+		pet.age = 30;
+		expect(() => pet.walk()).toThrow('Your pet is no longer alive :(');
+	});
+});
+
+describe('feed', () => {
+	it('decreases pet hunger level by 3', () => {
+		pet.hunger = 5;
+		pet.feed();
+		expect(pet.hunger).toEqual(2);
+	});
+	it('stops hunger going less than 0', () => {
+		pet.hunger = 1;
+		pet.feed();
+		expect(pet.hunger).toEqual(0);
+	});
+	it('throws an error if the pet is not alive', () => {
+		pet.age = 30;
+		expect(() => pet.feed()).toThrow('Your pet is no longer alive :(');
+	});
+});
+
+describe('checkUp', () => {
+	it('return "I need a walk" if fitness less than 4 and hunger less than 5', () => {
+		pet.hunger = 1;
+		pet.fitness = 1;
+		expect(pet.checkUp()).toEqual('I need a walk');
+	});
+	it('return "I am hungry" if hunger greater than 4 and fitness greater than 3', () => {
+		pet.hunger = 5;
+		pet.fitness = 9;
+		expect(pet.checkUp()).toEqual('I am hungry');
+	});
+	it('return "I am hungry AND I need a walk" if hunger greater than 4 and fitness less than 4', () => {
+		pet.hunger = 9;
+		pet.fitness = 1;
+		expect(pet.checkUp()).toEqual('I am hungry AND I need a walk');
+	});
+
+	it('return "I feel great!" if hunger less than 5 and fitness greater than 3', () => {
+		expect(pet.checkUp()).toEqual('I feel great!');
+	});
+	it('returns message if the pet is not alive', () => {
+		pet.age = 30;
+		expect(pet.checkUp()).toEqual('Your pet is no longer alive :(');
+	});
+});
+
+describe('isAlive', () => {
```

Comment:
All these tests should not be named with hard coded numbers as here, replacing them with something like 'returns false if fitness level below minimum' would make for better more maintainable code